src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Pal.Unix/ExportStoreProvider.cs

```diff
@@ -0,0 +1,200 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Security.Cryptography;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Win32.SafeHandles;
+
+namespace Internal.Cryptography.Pal
+{
+    internal class ExportStoreProvider : IStorePal
+    {
+        private static readonly SafeEvpPKeyHandle InvalidPKeyHandle = new SafeEvpPKeyHandle(IntPtr.Zero, false);
+
+        private ICertificatePal _singleCertPal;
+        private X509Certificate2Collection _certs;
+
+        internal ExportStoreProvider(ICertificatePal singleCertPal)
+        {
+            _singleCertPal = singleCertPal;
+        }
+
+        internal ExportStoreProvider(X509Certificate2Collection certs)
+        {
+            _certs = certs;
+        }
+
+        public void Dispose()
+        {
+            // Don't dispose any of the resources, they're still owned by the caller.
+            _singleCertPal = null;
+            _certs = null;
+        }
+
+        private static void AssertExportOnly()
```

Comment:
I would use the verb "Throw" rather than "Assert" in the name as (1) you're not actually checking a condition here and (2) there's no other indication on the caller side that this method never returns.